### PR TITLE
igmpproxy: update to version 0.3

### PR DESCRIPTION
**Pull request by ja-pa**
This PR updates igmpproxy to version 0.3 [Changelog](https://github.com/pali/igmpproxy/releases/tag/0.3)

cc @pali

### CODE REVIEW

File: package/network/services/igmpproxy/Makefile [hunk 1]
```diff
@@ -39,13 +39,7 @@ define Package/igmpproxy/conffiles
...
 endef
 
-TARGET_CFLAGS += -Dlog=igmpproxy_log
-
-define Build/Compile
-	$(MAKE) -C $(PKG_BUILD_DIR)/src \
-		CC="$(TARGET_CC)" \
-		CFLAGS="$(TARGET_CFLAGS) -std=gnu99"
-endef
+TARGET_CFLAGS += -Dlog=igmpproxy_log -std=gnu99
```
Comment (pali) [inline package/network/services/igmpproxy/Makefile:42]: `-std=gnu99` should not be required as `./configure` automatically adds it if needed:
https://github.com/pali/igmpproxy/blob/0.3/configure.ac#L6

And I'm not sure what is `-Dlog=igmpproxy_log` doing... igmpproxy does not have any `log` function.

Comment (ja-pa) [reply package/network/services/igmpproxy/Makefile]: Thx. It looks like leftovers related to the previous version. I removed it and it compiles and works ok on Omnia master.

**Comment by aparcar** [comment]
Could this move to packages.git?

**Comment by ja-pa** [comment]
> Could this move to packages.git?

cc @nbd168

**Comment by adschm** [comment]
Missing commit message. Please make sure you also explain the reasons for the code changes.

**Comment by CodeFetch** [comment]
@aparcar I think this is a rather elementary feature for a router and should therefore not be moved to the community packages repository... OpenWrt's focus is too much on routers, so I agree it would be nice to not have it in the main repository, but there needs to be a new policy and maybe a new repository. Just my two cents...

**Comment by bobafetthotmail** [comment]
> @aparcar I think this is a rather elementary feature for a router and should therefore not be moved to the community packages repository... OpenWrt's focus is too much on routers, so I agree it would be nice to not have it in the main repository, but there needs to be a new policy and maybe a new repository. Just my two cents...

Seems like you think the "move to packages" means "this package is not useful". This is not true.

The move to package feed is to offload review of basic changes and update churn like this to the bigger pool of developers with commit access to the package repository.

Way too much package updates and other stuff remains stuck in limbo for many months because nobody of the core devs have time for them.

The end goal is to have only build system, core packages that are installed by default and device-specific pakcages and patches in the core OpenWrt repo.

**Comment by CodeFetch** [comment]
> The end goal is to have only build system, core packages that are installed by default and device-specific pakcages and patches in the core OpenWrt repo.

I'm aware of that. I said `I think this is a rather elementary feature for a router` with which I meant "I'd rather consider adding this as a default package than moving it to the community repository"

**Comment by bobafetthotmail** [comment]
> I'm aware of that. I said `I think this is a rather elementary feature for a router` with which I meant "I'd rather consider adding this as a default package than moving it to the community repository"
> 

I already told you, this is not a "project focus" problem where someone decides to bikeshed packages here or there based on his own personal definition of what is "fit for a router" and what is not.

This is a way to mitigate the very real technical problems with reviewing and merging contributions. Anything non-essential (i.e. not installed by default at this point, give or take) is liable to getting moved to community-maintained repos to remove the load on core repo developers, and allow faster updates of said packages that now don't need to wait months for someone to even show up.

> Edit: To make clear: I don't want it as a default package either. Maybe in the future we could add a number of default "router" packages which are being selected when building router devices.

Doing this in the build system is way too inflexible, as you also said. (Was tried in the past with the "nas" package profile, went poorly, was more or less scrapped). it adds a lot of work and will still upset a lot of people that wil get bloated images with stuff they don't want or need. Or you are exploding the storage requirements on download servers to store many different flavors of the firmware images. Plus you get the main issue that the "moving packages to community maintained repos" is solving, that core developers are swamped with contributions and can't review and merge fast enough.

I think the future is a "build-to-order" system like the "attended sysupgrade" by aparcar where the server just uses the ImageBuilder to assemble a firmware image with whatever packages you have installed in your system. This allows everyone to get a customized build of OpenWrt for his device without tripping over all kinds of real-life limitations.
https://github.com/aparcar/asu

**Comment by CodeFetch** [comment]
@bobafetthotmail My idea was just to think about introducing a new repository for OpenWrt packages which are essential for OpenWrt "router" purposes. OpenWrt has great package feed support. The LEDE split was done to get more order into this and igmpproxy is a good example of a package that "router" users might expect from a router to support out of the box. While packages in the OpenWrt main repository are working great from my experience community packages are often not maintained well. Therefore I'd like to keep a means of differentiation between them. Honestly I don't think we'll have a new repository soon as there are still OpenWrt feeds which are running on the old git instance. But I don't have a strong opinion on this. As I said: `Just my two cents...`

**Comment by bobafetthotmail** [comment]
> @bobafetthotmail My idea was just to think about introducing a new repository for OpenWrt packages which are essential for OpenWrt "router" purposes. OpenWrt has great package feed support. The LEDE split was done to get more order into this and igmpproxy is a good example of a package that "router" users might expect from a router to support out of the box. While packages in the OpenWrt main repository are working great from my experience community packages are often not maintained well. Therefore I'd like to keep a means of differentiation between them.

What's the point of moving stuff to another repo if the same people would have to check it and maintain it there too. I would see that as bikeshedding.

**Comment by rsalvaterra** [comment]
> I'm aware of that. I said `I think this is a rather elementary feature for a router` with which I meant "I'd rather consider adding this as a default package than moving it to the community repository"

I agree with @aparcar, igmpproxy could just as well live in the community packages repository (and including it in the default images makes no sense). Yes, it's very useful when you need it, but that applies to basically any other package. It's not essential for basic routing functionality. Of course, if you plan on configuring your router for, say, an IPTV service provided by your ISP, sure, you'll probably need igmpproxy. But that's hardly "basic routing functionality", is it?

**Comment by ja-pa** [comment]
@adschm I updated commit message

**Comment by hauke** [comment]
Thank you for the pull request, it was added to master.
I think this should go to the package feed, but this is unrelated to these changes.

**Pull request closed without merging**